File: src/service/MainService.java
package service;

import entity.ParkingLot;

import java.util.ArrayList;
import java.util.Scanner;

public class MainService {
    public static void run() {
        Scanner scanner = new Scanner(System.in);
        boolean repeat = true;

        System.out.println("Parking lot system\n");
        System.out.println("Creating parking lot,");
        System.out.println("Please input lot's capacity");

        ParkingLot lot = new ParkingLot(scanner.nextInt(),new ArrayList<>());
        System.out.println("Created a parking lot with "+lot.getSpace()+" slots");

        ParkingService parkingService = new ParkingService(lot);

        while (repeat) {
            try {
                System.out.println("\nManagements");
                System.out.println("1. Check-in vehicle");
                System.out.println("2. Check-out vehicle");
                System.out.println("3. Parking lot status");
                System.out.println("4. Search vehicles");
                System.out.println("5. Exit system");
                int selectMenu = scanner.nextInt();
                scanner.nextLine();


                switch (selectMenu) {
                    case 1:
                        System.out.println("input vehicle details with format ([plateNumber] [color] [mobil/motor])");
                        System.out.println("(ex: A-1234-YZ red mobil)");
                        String detail = scanner.nextLine();
                        parkingService.park(detail);
                        break;
                    case 2:
                        System.out.println("input slot number");
                        int slot = scanner.nextInt();
                        scanner.nextLine();
                        parkingService.leave(slot);
                        break;
                    case 3:
                        parkingService.status();
                        break;
                    case 4:
                        System.out.println("Search:");
                        System.out.println("1. Plate number by color");
                        System.out.println("2. Slot number by color");
                        System.out.println("3. Plate with odd number");
                        System.out.println("4. Plate with even number");
                        System.out.println("5. Count cars");
                        System.out.println("6. Count motorcycles");
                        System.out.println("7. Slot number by Plate number");
                        System.out.println("8. Back");
                        int searchMenu = scanner.nextInt();
                        scanner.nextLine();
                        String searchParam="";
                        switch (searchMenu) {
                            case 1,2:
                                System.out.println("input color: ");
                                searchParam = scanner.nextLine();
                                break;
                            case 7:
                                System.out.println("input plate number: ");
                                searchParam = scanner.nextLine();
                                break;
                            default:
                                searchParam = null;
                                break;
                        }
                        parkingService.search(searchMenu,searchParam);
                        break;
                    case 5:
                        System.out.println("Quitting system");
                        repeat = false;
                        break;
                    default:
                        System.out.println();
                        System.out.println("Incorrect input");
                        break;
                }
            } catch (Exception e) {
                System.out.println();
                System.out.println("Incorrect input");
                scanner.nextLine();
            }
        }
    }
}

File: dotnet-version/ParkingSystemDotnet/Services/MainService.cs
using System;
using System.Collections.Generic;
using ParkingSystemDotnet.Services;
using ParkingSystemDotnet.Models;

namespace ParkingSystemDotnet
{
    public class MainService
    {
        public static void Run()
        {
            var scanner = Console.In;
            bool repeat = true;

            Console.WriteLine("Parking lot system\n");
            Console.WriteLine("Creating parking lot,");
            Console.WriteLine("Please input lot's capacity");

            int capacity = int.Parse(Console.ReadLine() ?? "0");
            var lot = new ParkingLot(capacity, new List<Vehicle>());
            Console.WriteLine($"Created a parking lot with {lot.Space} slots");

            var parkingService = new ParkingService(lot);

            while (repeat)
            {
                try
                {
                    Console.WriteLine("\nManagements");
                    Console.WriteLine("1. Check-in vehicle");
                    Console.WriteLine("2. Check-out vehicle");
                    Console.WriteLine("3. Parking lot status");
                    Console.WriteLine("4. Search vehicles");
                    Console.WriteLine("5. Exit system");

                    int selectMenu = int.Parse(Console.ReadLine() ?? "0");

                    switch (selectMenu)
                    {
                        case 1:
                            Console.WriteLine("Input vehicle details with format ([plateNumber] [color] [mobil/motor])");
                            Console.WriteLine("(ex: A-1234-YZ red mobil)");
                            string detail = Console.ReadLine();
                            parkingService.Park(detail);
                            break;

                        case 2:
                            Console.WriteLine("Input slot number");
                            int slot = int.Parse(Console.ReadLine() ?? "0");
                            parkingService.Leave(slot);
                            break;

                        case 3:
                            parkingService.Status();
                            break;

                        case 4:
                            Console.WriteLine("Search:");
                            Console.WriteLine("1. Plate number by color");
                            Console.WriteLine("2. Slot number by color");
                            Console.WriteLine("3. Plate with odd number");
                            Console.WriteLine("4. Plate with even number");
                            Console.WriteLine("5. Count cars");
                            Console.WriteLine("6. Count motorcycles");
                            Console.WriteLine("7. Slot number by Plate number");
                            Console.WriteLine("8. Back");

                            int searchMenu = int.Parse(Console.ReadLine() ?? "0");
                            string searchParam = string.Empty;

                            switch (searchMenu)
                            {
                                case 1:
                                case 2:
                                    Console.WriteLine("Input color: ");
                                    searchParam = Console.ReadLine();
                                    break;
                                case 7:
                                    Console.WriteLine("Input plate number: ");
                                    searchParam = Console.ReadLine();
                                    break;
                                default:
                                    searchParam = null;
                                    break;
                            }

                            parkingService.Search(searchMenu, searchParam);
                            break;

                        case 5:
                            Console.WriteLine("Quitting system");
                            repeat = false;
                            break;

                        default:
                            Console.WriteLine("\nIncorrect input");
                            break;
                    }
                }
                catch (Exception)
                {
                    Console.WriteLine("\nIncorrect input");
                }
            }
        }
    }
}